src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -299,8 +299,7 @@ public function sendContent()
 
         fclose($out);
         fclose($file);
-
-        if ($this->deleteFileAfterSend) {
+        if ($this->deleteFileAfterSend && file_exists($this->file->getPathname())) {
```

Comment:
doenst `fopen` already fail in this case?

Reply:
The file can be deleted after fopen() while it is being sent to the user.

Reply:
right, but if the file does not exists `fopen` would already trigger `ErrorException` no? see https://3v4l.org/64nZg

if it does exist, there should be no error anyway: https://3v4l.org/jraIM

Reply:
got ya :)

Reply:
But the file could also be removed between the calls to `file_exists()` and `unlink()`. Maybe it would be better to silence the delete operation instead.

Reply:
Downloading is a much longer process than `file_exists()` )

Reply:
Also, I do not think that `@unlink($this->file->getPathname());` would be a good solution here, it will ignore all errors, and not just 'No such file or directory'.
What else needs to be done for this to be merged ?